.github/workflows/flutter_pub_publish.yml

```diff
@@ -55,11 +52,18 @@ jobs:
       - name: 📦 Install Dependencies
         run: flutter pub get
 
-      - name: 🔐 Setup Pub Credentials
+      - name: 🔐 Setup Pub Credentials\
+        if: ${{ secrets.pub_credentials != '' }}
         run: |
           mkdir -p $XDG_CONFIG_HOME/dart
           echo '${{secrets.pub_credentials}}' > "$XDG_CONFIG_HOME/dart/pub-credentials.json"
 
+      - name: 🎯 Setup Dart
+        if: ${{ secrets.pub_credentials == '' }}
+        uses: dart-lang/setup-dart@v1
+        with:
+          sdk: ${{inputs.dart_sdk}}
```

Comment:
Because that's what actually set's up the auth token with Pub.

Reply:
As talked about in standup, we're going to open an issue on https://github.com/subosito/flutter-action to see if there is a way to have this done within that instead of doing a double setup of Dart in the workflow overall.

Reply:
Ticket created here https://github.com/subosito/flutter-action/issues/330